app/import_functions_test.go

```diff
@@ -1932,12 +1932,403 @@ func TestImportImportPost(t *testing.T) {
+		data := &PostImportData{
+			Team:     &teamName,
+			Channel:  &channelName,
+			User:     &username,
+			Message:  ptrStr("Hello"),
+			CreateAt: ptrInt64(model.GetMillis()),
+		}
+		err := th.App.importMultiplePosts([]*PostImportData{data}, true)
+		assert.Nil(t, err)
+		AssertAllPostsCount(t, th.App, initialPostCount, 0, team.Id)
+	})
+
+	t.Run("Try adding an invalid post in apply mode", func(t *testing.T) {
+		data := &PostImportData{
+			Team:     &teamName,
+			Channel:  &channelName,
+			User:     &username,
+			CreateAt: ptrInt64(model.GetMillis()),
+		}
+		err := th.App.importMultiplePosts([]*PostImportData{data}, false)
+		assert.NotNil(t, err)
+		AssertAllPostsCount(t, th.App, initialPostCount, 0, team.Id)
+	})
+
+	t.Run("Try adding a valid post with invalid team in apply mode", func(t *testing.T) {
+		data := &PostImportData{
+			Team:     ptrStr(model.NewId()),
+			Channel:  &channelName,
+			User:     &username,
+			Message:  ptrStr("Message"),
+			CreateAt: ptrInt64(model.GetMillis()),
+		}
+		err := th.App.importMultiplePosts([]*PostImportData{data}, false)
+		assert.NotNil(t, err)
+		AssertAllPostsCount(t, th.App, initialPostCount, 0, team.Id)
+	})
+
+	t.Run("Try adding a valid post with invalid channel in apply mode", func(t *testing.T) {
+		data := &PostImportData{
+			Team:     &teamName,
+			Channel:  ptrStr(model.NewId()),
+			User:     &username,
+			Message:  ptrStr("Message"),
+			CreateAt: ptrInt64(model.GetMillis()),
+		}
+		err := th.App.importMultiplePosts([]*PostImportData{data}, false)
+		assert.NotNil(t, err)
+		AssertAllPostsCount(t, th.App, initialPostCount, 0, team.Id)
+	})
+
+	t.Run("Try adding a valid post with invalid user in apply mode", func(t *testing.T) {
+		data := &PostImportData{
+			Team:     &teamName,
+			Channel:  &channelName,
+			User:     ptrStr(model.NewId()),
+			Message:  ptrStr("Message"),
+			CreateAt: ptrInt64(model.GetMillis()),
+		}
+		err := th.App.importMultiplePosts([]*PostImportData{data}, false)
+		assert.NotNil(t, err)
+		AssertAllPostsCount(t, th.App, initialPostCount, 0, team.Id)
+	})
+
+	t.Run("Try adding a valid post in apply mode", func(t *testing.T) {
+		data := &PostImportData{
+			Team:     &teamName,
+			Channel:  &channelName,
+			User:     &username,
+			Message:  ptrStr("Message"),
+			CreateAt: &time,
+		}
+		err := th.App.importMultiplePosts([]*PostImportData{data}, false)
+		assert.Nil(t, err)
+		AssertAllPostsCount(t, th.App, initialPostCount, 1, team.Id)
+
+		// Check the post values.
+		posts, err := th.App.Srv.Store.Post().GetPostsCreatedAt(channel.Id, time)
+		require.Nil(t, err)
+
+		require.Len(t, posts, 1, "Unexpected number of posts found.")
+
+		post := posts[0]
+		postBool := post.Message != *data.Message || post.CreateAt != *data.CreateAt || post.UserId != user.Id
+		require.False(t, postBool, "Post properties not as expected")
+	})
+
+	t.Run("Update the post", func(t *testing.T) {
```

Comment:
I'm reformating tests here only, but I think we can do better, I changed the message to other thing, to properly test it. The idea is, when you import a post, if the channelId and the exact timestamp is the same, we consider that you are updating the post instead of creating a new one. This is the idea here, update the post using the import tool.

Reply:
I see. Thanks.